### PR TITLE
bpo-27062: add `__all__` to inspect module

### PR DESCRIPTION
**Pull request by kumaraditya303**
https://bugs.python.org/issue27062

### CODE REVIEW

File: Lib/inspect.py [hunk 1]
```diff
@@ -32,7 +32,18 @@
...
+            'isgeneratorfunction', 'iscoroutinefunction', 'isasyncgenfunction', 'isasyncgen', 
+            'isgenerator', 'iscoroutine', 'isawaitable', 'istraceback', 'isframe', 'iscode',
+            'isbuiltin', 'isroutine', 'isabstract', 'getmembers', 'getmembers_static', 
+            'classify_class_attrs', 'getmro', 'unwrap', 'indentsize', 'getdoc', 'cleandoc', 
+            'getfile', 'getmodulename', 'getsourcefile', 'getabsfile', 'getmodule', 'findsource', 
+            'getcomments', 'getblock', 'getsourcelines', 'getsource', 'walktree', 'getclasstree',
+            'getargs', 'getfullargspec', 'getargvalues', 'formatannotation', 'formatannotationrelativeto', 
+            'formatargvalues', 'getcallargs', 'getclosurevars', 'getframeinfo', 'getlineno', 'getouterframes', 
+            'getinnerframes', 'currentframe', 'stack', 'trace', 'getattr_static', 'getgeneratorstate', 
+            'getgeneratorlocals', 'getcoroutinestate', 'getcoroutinelocals', 'signature']
```
Comment (iritkatriel) [inline Lib/inspect.py]: It will be easier to maintain if it was sorted.

Comment (kumaraditya303) [reply Lib/inspect.py]: Sorted.

File: Lib/inspect.py [hunk 2]
```diff
@@ -32,7 +32,20 @@
 
 __author__ = ('Ka-Ping Yee <ping@lfw.org>',
               'Yury Selivanov <yselivanov@sprymix.com>')
-
+__all__ = ['classify_class_attrs', 'cleandoc', 'currentframe', 'findsource',
```
Comment (tiran) [inline Lib/inspect.py]: I'm not a fan of this indention style. Please use

```suggestion
__all__ = [
    'classify_class_attrs', 'cleandoc', 'currentframe', 'findsource',
    ...
]
```

**Review by tiran — DISMISSED** [review]
This changeset needs a test to verify that ``__all__`` does not regress.

**Review by asvetlov — DISMISSED** [review]
`__all__` is for public API.
`Parameter`, `Signature`, and `BoundArguments` should be enumerated along with other names.

File: Lib/test/test_inspect.py [hunk 3]
```diff
@@ -107,6 +107,10 @@ def istest(self, predicate, exp):
                 continue
             self.assertFalse(other(obj), 'not %s(%s)' % (other.__name__, exp))
 
+    def test___all__(self):
+        self.assertGreater(len(inspect.__all__), 0)
+        self.assertIn("signature", inspect.__all__)
+
```
Comment (iritkatriel) [inline Lib/test/test_inspect.py:112]: Have a look at test.support.check__all__

**Comment by kumaraditya303** [comment]
I have made the requested changes; please review again

File: Lib/test/test_inspect.py (hunk 3 above)
Comment (kumaraditya303) [reply Lib/test/test_inspect.py:112]: That cannot be used as inspect module defines k and many others as global

Comment (iritkatriel) [reply Lib/test/test_inspect.py:112]: Can you specify that as not_exported?
https://github.com/python/cpython/blob/a6c3b0faa1d55e36539caf19bd3bcf1dea12df84/Lib/test/support/__init__.py#L1496

Comment (kumaraditya303) [reply Lib/test/test_inspect.py:112]: done

File: Lib/inspect.py [hunk 4]
```diff
@@ -32,7 +32,106 @@
...
+    "isgeneratorfunction",
+    "isgetsetdescriptor",
+    "ismemberdescriptor",
+    "ismethod",
+    "ismethoddescriptor",
+    "ismodule",
+    "isroutine",
+    "istraceback",
+    "signature", 
+    "CORO_CLOSED",
```
Comment (AlexWaygood) [inline Lib/inspect.py]: This list is no longer alphabetised (might it be worth adding a test to ensure it stays alphabetised, or would that be overkill?)

Comment (kumaraditya303) [reply Lib/inspect.py]: sorted, test would be overkill

**Review by ambv — APPROVED** [review]

File: Lib/inspect.py [hunk 5]
```diff
@@ -32,7 +32,106 @@
 
 __author__ = ('Ka-Ping Yee <ping@lfw.org>',
               'Yury Selivanov <yselivanov@sprymix.com>')
-
+
```
Comment (aisk) [inline Lib/inspect.py:35]: There was some invisible spaces here. I think this is not expected.

**Pull request merged**